src/common/translation.cpp

```diff
@@ -1546,17 +1546,16 @@ wxString wxTranslations::GetBestTranslation(const wxString& domain,
     return lang;
 }
 
+WX_DECLARE_HASH_SET(wxString, wxStringHash, wxStringEqual,
+    wxLocaleUntranslatedStrings);
 
 /* static */
 const wxString& wxTranslations::GetUntranslatedString(const wxString& str)
 {
-    wxLocaleUntranslatedStrings& strings = wxThreadInfo.untranslatedStrings;
-
-    wxLocaleUntranslatedStrings::iterator i = strings.find(str);
-    if ( i == strings.end() )
-        return *strings.insert(str).first;
-
-    return *i;
+    static wxCriticalSection s_cs;
```

Comment:
This is only MT-safe in C++11, in C++98 we'd need to make it global. Not sure if it's worth changing this...